src/realm/bplustree.hpp

```diff
@@ -568,6 +566,13 @@ class BPlusTree : public BPlusTreeBase {
         m_root->bptree_traverse(func);
     }
 
+    void split_if_needed()
```

Comment:
Why is this needed? I am under the impression that `BPlusTree::insert()` will split automatically as needed?

Reply:
If the current node is just a plain array, then it should be split into a BPlusTree. I am open for another name - I struggled a bit myself to find one.

Reply:
So this is an optimization that defers creating a tree until we reach more than 1000 links and then does the transformation on the fly rather than in the migration function of the file format. If that is correct, we should keep current performance the same for fewer than 1000 backlinks?

Reply:
If the size is below 1000, then a BPlusTree is just a plain array, so no need for any transformation. But the BPlusTree implementation is just slower than a plain array. I did not want to create two code paths - one for sizes under 1000 and one for sizes over.